x/payment/client/testutil/integration_test.go

```diff
@@ -142,7 +142,7 @@ func (s *IntegrationTestSuite) TestSubmitWirePayForData() {
 						require.NoError(err)
 						msgSize, err := strconv.ParseUint(e.GetAttributes()[1].GetValue(), 10, 64)
 						require.NoError(err)
-						s.Equal(uint64(0), msgSize%consts.ShareSize, "Message length should be multiples of const.ShareSize=%v", consts.ShareSize)
+						require.Equal(uint64(consts.ShareSize), msgSize, "Message length should be %v", consts.ShareSize)
```

Comment:
while this specific message is the size of the `consts.ShareSize`, not all messages will be this size. The only reason that it is this size is because it gets padded. Ideally, we would have a function that calculated the padded length of the message being used in the test, and then use that to compare with the result here. I think it would also be okay to simply add a field in the test case to communicate the expected size.

Reply:
Hey @evan-forbes, thanks for the review, appreciate it!

I added the padMessage function, which adds the padding to the provided message and uses the result for comparison on the test.